### PR TITLE
Support delete varChar value

### PR DESCRIPTION
**Pull request by xige-16**
fix issue: #15888 
/kind: feature
Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

File: internal/datanode/flow_graph_delete_node.go [hunk 1]
```diff
@@ -161,7 +147,7 @@ func (dn *deleteNode) bufferDeleteMsg(msg *msgstream.DeleteMsg, tr TimeRange) er
 			delData.Pks = append(delData.Pks, pks[i])
 			delData.Tss = append(delData.Tss, tss[i])
 			log.Debug("delete",
-				zap.Int64("primary key", pks[i]),
+				zap.Any("primary key", pks[i]),
```
Comment (DragonDriver) [inline internal/datanode/flow_graph_delete_node.go:150]: Better to implement `String()` for `PrimaryKey`.

File: internal/querynode/segment.go [hunk 2]
```diff
@@ -531,24 +540,40 @@ func (s *Segment) segmentDelete(offset int64, entityIDs *[]UniqueID, timestamps
...
+		for index, entity := range entityIDs {
+			int64Pks[index] = entity.(*int64PrimaryKey).Value
+		}
+		var cEntityIdsPtr = (*C.int64_t)(&int64Pks[0])
+		status := C.Delete(s.segmentPtr, cOffset, cSize, cEntityIdsPtr, cTimestampsPtr)
+		if err := HandleCStatus(&status, "Delete failed"); err != nil {
+			return err
+		}
+	case schemapb.DataType_VarChar:
+		//TODO::
```
Comment (DragonDriver) [inline internal/querynode/segment.go:574]: should we use the marshaled StringArray?

Comment (xige-16) [reply internal/querynode/segment.go:574]: agree

**Pull request merged**